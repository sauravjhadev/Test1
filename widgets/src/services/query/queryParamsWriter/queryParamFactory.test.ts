import fs from 'fs';

import QueryParamFactory from './queryParamFactory';

const configData = JSON.parse(fs.readFileSync('src/services/__fixtures__/config.json', 'utf8').toString());
const builderData = JSON.parse(fs.readFileSync('src/services/__fixtures__/queryParamsBuilder.json', 'utf8').toString());
const queryData = fs.readFileSync('src/services/__fixtures__/query.graphql', 'utf8').toString();

describe('queryParamFactory', () => {
    it('correctly loads query params', () => {
        const factory = new QueryParamFactory(queryData, builderData, configData);
        expect(factory.build()).toMatchSnapshot();
    });
});
