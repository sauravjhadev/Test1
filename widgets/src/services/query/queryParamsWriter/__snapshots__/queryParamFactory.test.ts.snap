// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['queryParamFactory correctly loads query params 1'] = `
Object {
  "name": "Fred",
  "productId": 5,
  "productIds": Array [
    1,
  ],
}
`;
